activesupport/test/cache/behaviors/cache_store_behavior.rb

```diff
@@ -21,8 +21,9 @@ def test_fetch_without_cache_miss
   end
 
   def test_fetch_with_cache_miss
-    assert_called_with(@cache, :write, ["foo", "baz", @cache.options]) do
-      assert_equal "baz", @cache.fetch("foo") { "baz" }
+    key = ActiveSupport::Cache::KeyMaster.new("foo")
+    assert_called_with(@cache, :write, [key, "baz", @cache.options]) do
+      assert_equal "baz", @cache.fetch(key) { "baz" }
```

Comment:
Is this meant to incidentally test with the new key instances? Or to demonstrate their usage?

Think we'll likely want to leave these tests demonstrating typical usage with string keys and introduce tests asserting that the cache stores behave as expected with new key instances.

Reply:
The reason I modified this is the `assert_called_with` uses the exact value and in this case `write` receives an instance of `KeyMaster`. Previously it received an instance of a string. I couldn't find another way to re-write this without directly passing in the object. Otherwise, if we pass in a regular string it gets converted to the key object so the mock fails. If you've got ideas here i'm open to suggestion.

Reply:
Got it! 👍